### PR TITLE
Rdrp 994 clone folders

### PR DESCRIPTION
**Pull request by zorge69**
## Created all input and output folders in /bat/res_dev/ 
Updated the make_dir to use the config explicitly
Fixed a bug in the usage of tree_to_list.
Uploaded anonymised snapshot to /res_dev/anonymised
Uploaded anonymised backdata to /bat/res_dev/project_data/2021_surveys/BERD/06_imputation/backdata_output
Uploaded all mappers to /bat/res_dev/project_data/2023_surveys/mappers/v1 (ultfoc is anonymised)
Created log folder /bat/res_dev/project_data/logs/run_logs

# The program still does not run. We need to merge the singleton class to develop. We need to undo the explicit passage of config argument in rd_make_dirs

*Insert detailed bullet points about your changes here!*

*Insert any instructions to help the reviewer, e.g. "install new requirements from `requirements.txt`"*

*Let the reviewer know what data files are needed (and if applicable, where they are to be found)

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes #

#### Code

- [ ] **Code runs** The code runs on my machine and/or CDSW
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `requirements.txt`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/user_config.yaml [hunk 1]
```diff
@@ -63,18 +63,18 @@ hdfs_paths:
   freezing_additions_path: "02_freezing/freezing_updates"
   freezing_amendments_path: "02_freezing/freezing_updates"
   # Imputation and outliers input paths
-  backdata_path: ""
+  backdata_path: "2021_surveys/BERD/06_imputation/backdata_output/2021_backdata_published_v347_anon.csv"
```
Comment (AnneONS) [inline src/user_config.yaml]: This needs to go in 2022_surveys so we can run the 2023 pipeline.

File: src/user_config.yaml [hunk 2]
```diff
@@ -63,18 +63,18 @@ hdfs_paths:
...
+  backdata_path: "2022_surveys/BERD/06_imputation/backdata_output/2022_backdata_anon.csv"
   manual_imp_trim_path: "06_imputation/manual_trimming/trimming_qa_2023-11-27_v359.csv"
   manual_outliers_path: "07_outliers/manual_outliers/manual_outlier_2023-08-29_v67.csv"
   # Construction paths
   all_data_construction_file_path:  "04_construction/manual_construction/test_construction_file - Copy.csv"
   postcode_construction_file_path: "04_construction/manual_construction/test_postcode_construction_file.csv"
   construction_file_path_ni:  "04_construction/manual_construction/test_construction_file.csv"
   # postcode paths
-  postcode_masterlist: "/ons/rdbe_dev/mappers/2023/ONSPD_NOV_2022_UK.csv"
+  postcode_masterlist: "/bat/res_dev/mappersONSPD_NOV_2022_UK.csv"
```
Comment (AnneONS) [inline src/user_config.yaml]: this path doesn't look right....

**Review by AnneONS — APPROVED** [review]
I have removed config from mk_dirs, and I have also corrected the tree, as it was a bit off. I've checked by creating folders in my own area- it does run and does what it's meant to.

File: src/user_config.yaml (hunk 1 above)
Comment (AnneONS) [reply src/user_config.yaml]: I have corrected this one.

File: src/user_config.yaml (hunk 2 above)
Comment (zorge69) [reply src/user_config.yaml]: Fixed an error. Uploaded a small extract from the postcode lookup file.

**Pull request merged**